src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -143,6 +143,11 @@ class Task extends React.Component<Props, State> {
 
     handleEditSubmitSuccess = () => {
         this.setState({ isEditing: false });
+        const { onModalClose } = this.props;
```

Comment:
Added this because when the modal is closed via the Edit flow, we aren't clearing the approverSelector state and it's still dirty. 

I fixed this bug via the create flow here: https://github.com/box/box-ui-elements/pull/1189/